File: app/src/test/java/com/example/benchmarks/OperationsCollectionsUnitTest.java
package com.example.benchmarks;

import static org.junit.Assert.*;

import com.example.benchmarks.models.Benchmark;
import com.example.benchmarks.models.BenchmarkItem;
import com.example.benchmarks.models.OperationsCollections;

import org.junit.Test;

import java.util.List;

public class OperationsCollectionsUnitTest {

    private final Benchmark benchmark = new OperationsCollections();

    @Test
    public void testCreatingBenchmarkListReturnsNotNull() {
        final List<BenchmarkItem> testList = benchmark.createBenchmarkList(false);
        assertNotNull(testList);
    }

    @Test
    public void testCreatingBenchmarkListIsNotEmpty(){
        final List<BenchmarkItem> testList = benchmark.createBenchmarkList(false);
        assertTrue(testList.size() > 0);
    }

    @Test
    public void testMeasureTimeMoreThan0(){
        final List<BenchmarkItem> testList = benchmark.createBenchmarkList(false);
        for (BenchmarkItem item : testList) {
            assertTrue(benchmark.measureTime(5000, item) > 0);
        }
    }

    @Test
    public void testMeasureTimeThrowsException(){
        BenchmarkItem wrongItem = new BenchmarkItem(1, 1, 0L, false);
        assertThrows(RuntimeException.class, () -> benchmark.measureTime(10, wrongItem));
    }

    @Test
    public void testGetSpansCount(){
        assertEquals(3, benchmark.getSpansCount());
    }
}

File: app/src/test/java/com/example/benchmarks/models/benchmark/OperationsCollectionsUnitTest.kt
package com.example.benchmarks.models.benchmark

import org.junit.Assert.*
import org.junit.Test

class OperationsCollectionsUnitTest {

    private val benchmark: Benchmark = OperationsCollections()

    @Test
    fun testCreatingBenchmarkListReturnsNotNull() {
        val testList = benchmark.createBenchmarkList(false)
        assertNotNull(testList)
    }

    @Test
    fun testCreatingBenchmarkListIsNotEmpty() {
        val testList = benchmark.createBenchmarkList(false)
        assertTrue(testList.isNotEmpty())
    }

    @Test
    fun testMeasureTimeMoreThan0() {
        val testList = benchmark.createBenchmarkList(false)
        for (item in testList) {
            assertTrue(benchmark.measureTime(5000, item) > 0)
        }
    }

    @Test
    fun testMeasureTimeThrowsException() {
        val wrongItem = BenchmarkItem(1, 1, 0L, false)
        assertThrows(RuntimeException::class.java) { benchmark.measureTime(10, wrongItem) }
    }

    @Test
    fun testGetSpansCount() {
        assertEquals(3, benchmark.getSpansCount())
    }
}